File: tyama_icpc2007dA.cs
using System;

class icpc2007dA{
  static void Main(){
    int i, n, s, ma, mi, sum;
    while(true){
      ma=0; mi=1000; sum=0;
      n=int.Parse(Console.ReadLine());
      if(n==0)return;
      for(i=0;i<n;i++){
        s=int.Parse(Console.ReadLine());
        ma=ma>s?ma:s;
        mi=mi<s?mi:s;
        sum+=s;
      }
      Console.WriteLine((sum-ma-mi)/(n-2));
    }
  }
}

File: tyama_icpc2007dA.vb
module icpc2007dA
    sub Main(ByVal args() as String)
        dim i,n,s,ma,mi,sum as integer
        while(true)
          ma=0:mi=1000:sum=0
          n=Console.ReadLine()
          if n=0 then
            exit while
          end if
          for i=1 to n
            s=Console.ReadLine()
            ma=if(ma>s,ma,s) 'VB2008only: for VB2005 use if ma<s then \n ma=s \n end if
            mi=if(mi<s,mi,s)
            sum=sum+s
          next
          Console.WriteLine(int((sum-ma-mi)/(n-2)))
        end while
    end sub
end module